### PR TITLE
Bit shifting for unsigned integer doubling

### PR DESCRIPTION
**Pull request by irfanbozkurt**
## Description

Changes doubling logic for unsigned integers from addition to bit shifting

## Type of change

- [X] Optimization

### CODE REVIEW

**Comment by Oppen** [comment]
Any benchmarks for comparison?

**Comment by Oppen** [comment]
Answering myself from data in Slack:
> Results for doubling an unsigned integer 10000000 times and assigning:
> Time elapsed in double() is: 810.752416ms (OLD)
> Time elapsed in double() is: 548.382958ms (NEW, CLONES)
> Time elapsed in double_in_place() is: 353.955042ms (NEW, IN PLACE)

**Review by Oppen — APPROVED** [review]

File: math/src/unsigned_integer/element.rs [hunk 1]
```diff
@@ -597,9 +597,22 @@ impl<const NUM_LIMBS: usize> UnsignedInteger<NUM_LIMBS> {
         (UnsignedInteger { limbs }, carry > 0)
     }
 
-    /// Returns the double of `self`.
     pub fn double(a: &UnsignedInteger<NUM_LIMBS>) -> (UnsignedInteger<NUM_LIMBS>, bool) {
-        Self::add(a, a)
+        let mut cloned = (*a).clone();
```
Comment (Oppen) [inline math/src/unsigned_integer/element.rs:601]: This should fix clippy:
```suggestion
        let mut cloned = *a;
```

**Comment by Oppen** [comment]
@diegokingston I checked the failing pipeline. The tests pass fine, but codecov upload failed. I'm retrying to check if it's just a temporary service failure. Otherwise, I suspect it has to do with external contributors not being able to use our API key.

**Review by diegokingston — APPROVED** [review]

**Pull request merged**